drivers/dp/swdp_bitbang.c

```diff
@@ -668,7 +668,10 @@ static int sw_port_off(const struct device *dev)
 	}
 
 	if (config->reset.port) {
-		ret = gpio_pin_configure_dt(&config->reset, GPIO_DISCONNECTED);
+		ret = gpio_pin_configure_dt(&config->reset,
+					    IS_ENABLED(CONFIG_SWDP_MAINTAIN_RESET_DEASSERTED)
+						    ? GPIO_OUTPUT_ACTIVE
```

Comment:
If this option is enabled and the debug interface is disabled, the target SoC is kept under reset. This is not what the commit message describes. GPIO_OUTPUT_ACTIVE does not mean output high level.

I do not like this change, it is a workaround for the hardware design flaw.
Is it connected directly to the target without any buffers? No pull-up on nRESET?